core/src/main/java/org/ehcache/core/internal/service/ServiceLocator.java

```diff
@@ -205,10 +206,11 @@ private boolean hasUnstartedDependencies(Service service, Iterable<Service> unst
     return false;
   }
 
-  private boolean hasRunningDependencies(Service service, Iterable<Service> running) {
-    for (Class<? extends Service> dep : identifyTransitiveDependenciesOf(service.getClass())) {
-      for (Service s : running) {
-        if (dep.isInstance(s)) {
+  private boolean aRunningServiceStillDependsOnMe(Service service, Iterable<Service> running) {
+    for (Service runningService : running) {
+      Set<Class<? extends Service>> dependencies = identifyTransitiveDependenciesOf(runningService.getClass());
```

Comment:
I would say dependencyClasses and dependencyClass to remove the doubt that it is an actual dependency instance